### PR TITLE
Test: extend validate on errors on Runtime

### PR DESCRIPTION
**Pull request by eloycoto**
Currently the runtime test does not have the full validation of logs
messages as Kubernetes test. This adds al the methods for fail and
checks.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5086)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/helpers/cilium.go [hunk 1]
```diff
@@ -30,7 +30,6 @@ import (
 	"github.com/cilium/cilium/test/config"
```
Comment (ianvernon) [inline test/helpers/cilium.go:30]: Please proofread your commit messages. "al" --> "all"

File: test/helpers/cilium.go [hunk 2]
```diff
@@ -642,19 +640,38 @@ func (s *SSHMeta) ValidateNoErrorsOnLogs(duration time.Duration) {
...
 
+	defer func() {
+		// Keep the cilium logs for the given test in a separate file.
+		testPath, err := CreateReportDirectory()
+		if err != nil {
+			s.logger.WithError(err).Error("Cannot create report directory")
+			return
+		}
+		err = ioutil.WriteFile(
+			fmt.Sprintf("%s/cilium_test.log", testPath),
```
Comment (ianvernon) [inline test/helpers/cilium.go]: Make "cilium_test.log" a string variable / constant and reuse it.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Looks fine, mainly nitpicks.

File: test/helpers/cilium.go (hunk 1 above)
Comment (eloycoto) [reply test/helpers/cilium.go:30]: done

**Comment by eloycoto** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**